src/components/__tests__/countriesIntegration.test.js

```diff
@@ -0,0 +1,74 @@
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { Provider, useSelector } from 'react-redux';
+import { MemoryRouter } from 'react-router-dom';
+import Countries from '../Countries';
+import store from '../../redux/store';
+
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useSelector: jest.fn(),
+}));
+
```

Comment:
- [ ] Good job adding integration tests for your app with the react testing library, however, let's also add enough unit tests for the individual components like Cities, Countries, Info and States.

Reply:
Okay. Thanks for the feedback!